pallets/gear/src/manager/mod.rs

```diff
@@ -279,4 +280,32 @@ where
             );
         }
     }
+
+    fn send_signal(&mut self, message_id: MessageId, destination: ProgramId) {
+        let reserved = GasHandlerOf::<T>::system_unreserve(message_id)
+            .unwrap_or_else(|e| unreachable!("GasTree corrupted! {:?}", e));
+        if reserved != 0 {
+            log::debug!(
+                "Send signal issued by {} to {} with {} supply",
+                message_id,
+                destination,
+                reserved
+            );
+
+            // Creating signal message.
+            let trap_signal = SignalMessage::new(message_id, core_processor::ERR_STATUS_CODE)
+                .into_dispatch(destination)
+                .into_stored();
+
+            // Splitting gas for newly created reply message.
+            GasHandlerOf::<T>::split(message_id, trap_signal.id())
```

Comment:
Open issue with discussed optimisation (somehow use the parent node gas)

Reply:
#1743